# ÔN TẬP KIỂM TRA THỰC HÀNH THỐNG KÊ MÁY TÍNH

Import thư viện

In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

### 1. Đọc dữ liệu và cho biết dữ liệu có bao nhiêu dòng, bao nhiêu cột

Đáp án: (10 dòng, 5 cột)

In [2]:
df = pd.read_csv('ds_salaries.csv')

In [3]:
# Kiểm tra dữ liệu có bao nhiêu dòng, bao nhiêu cột
# Kích thước dữ liệu

df.shape

(3755, 12)

Dữ liệu có 3755 dòng và 12 cột

### 2. Cho biết thuộc tính nào là thuộc tính phân loại, thuộc tính nào là thuộc tính định lượng

In [4]:
# Kiểm tra kiểu dữ liệu của df
df.dtypes

work_year               int64
experience_years      float64
experience_level       object
employment_type        object
job_title              object
salary                  int64
salary_currency        object
salary_in_usd           int64
employee_residence     object
remote_ratio            int64
company_location       object
company_size           object
dtype: object

In [5]:
# Phân loại dữ liệu
categorical_attributes = df.select_dtypes(include=['object']).columns
numerical_attributes = df.select_dtypes(include=['int64', 'float64']).columns

In [7]:
# Hiển thị kết quả
print("Thuộc tính phân loại:", categorical_attributes)
print("Thuộc tính định lượng:", numerical_attributes)

Thuộc tính phân loại: Index(['experience_level', 'employment_type', 'job_title', 'salary_currency',
       'employee_residence', 'company_location', 'company_size'],
      dtype='object')
Thuộc tính định lượng: Index(['work_year', 'experience_years', 'salary', 'salary_in_usd',
       'remote_ratio'],
      dtype='object')


### 3. Trong dữ liệu có thông tin của bao nhiêu người làm ngành IT?

Đáp án: 3755 người

In [9]:
num_employees = df.shape[0]
print(f"Số lượng người làm ngành IT là: {num_employees} người")

Số lượng người làm ngành IT là: 3755 người


### 4. Đếm số nhân viên theo kinh nghiệm làm việc

In [21]:
experience_level_count = df["experience_level"].value_counts()
print(experience_level_count)

experience_level
SE    2516
MI     805
EN     320
EX     114
Name: count, dtype: int64


### 5. Đếm số nhân viên làm việc ở các quốc gia khác nhau

In [23]:
# Đếm số nhân viên đang làm việc tại mỗi quốc gia
employee_count_by_country = df['employee_residence'].value_counts()

# In kết quả ra màn hình
print(employee_count_by_country)

employee_residence
US    3004
GB     167
CA      85
ES      80
IN      71
      ... 
BA       1
AM       1
CY       1
KW       1
MT       1
Name: count, Length: 78, dtype: int64


### 6. Tổng giá trị lương thực lãnh của tất cả nhân viên làm ngành IT

Đáp án: 516576814 USD

In [12]:
total_salaries = df["salary_in_usd"].sum()
print(f"Tổng giá trị lương thực lãnh của tất cả nhân viên làm ngành IT là: {total_salaries} USD")

Tổng giá trị lương thực lãnh của tất cả nhân viên làm ngành IT là: 516576814 USD


### 7. Top 3 vị trí công việc phổ biến nhất

In [22]:
# Nhóm dữ liệu theo cột 'job_title' và đếm số lượng người làm trong mỗi nhóm
job_count = df['job_title'].value_counts()

# Lấy ra top 3 chức danh công việc có số lượng người làm nhiều nhất
top_3_jobs = job_count.head(3)

# In ra kết quả
print("Top 3 vị trí công việc phổ biến nhất")
print(top_3_jobs)

Top 3 công việc phổ biến nhất
job_title
Data Engineer     1040
Data Scientist     840
Data Analyst       612
Name: count, dtype: int64


### 8. Top 3 vị trí công việc có mức lương trung bình cao nhất

In [24]:
# Tính mức lương trung bình cho mỗi vị trí công việc
average_salary_by_job = df.groupby('job_title')['salary_in_usd'].mean()

# Sắp xếp dữ liệu theo mức lương trung bình giảm dần
sorted_data_salaries = average_salary_by_job.sort_values(ascending=False)

# Lấy top 3 vị trí công việc có mức lương trung bình cao nhất
top_3_jobs = sorted_data_salaries.head(3)

# In ra kết quả
print("Top 3 vị trí công việc có mức lương trung bình cao nhất")
print(top_3_jobs)

Top 3 vị trí công việc có mức lương trung bình cao nhất
job_title
Data Science Tech Lead    375000.0
Cloud Data Architect      250000.0
Data Lead                 212500.0
Name: salary_in_usd, dtype: float64


### 9. Khảo sát vị trí công việc ngành IT